srcalc/src/test/java/gov/va/med/srcalc/web/view/VariableEntryTest.java

```diff
@@ -36,4 +36,19 @@ public final void testDefaults()
         expected.put(vars.get(5).getKey(), VariableEntry.SPECIAL_NUMERICAL);
         assertEquals(expected, entry.getDynamicValues());
     }
+    
+    @Test
+    public final void testWithRetrievedValues()
+    {
+    	// Contains multiple variables, but only one DiscreteNumericalVariable.
+        final List<AbstractVariable> vars = SampleObjects.sampleVariableList();
+        
+        // Behavior verification.
+        final VariableEntry entry = VariableEntry.withRetrievedValues(vars, SampleObjects.dummyPatient());
+        final HashMap<String, String> expected = new HashMap<>();
+        expected.put(vars.get(1).getKey(), "Male");
```

Comment:
Tight coupling to `dummyPatient()`. I recommend comparing to the patient data itself, e.g.:

`expected.put(vars.get(1).getKey(), patient.getGender());`

But getGender() will return "M" though the multiselect value is "Male". Hmm, let's talk about this.